CMakeLists.txt

```diff
@@ -841,15 +841,18 @@ if(EVENT__HAVE_EVENT_PORTS)
 endif()
 
 if (NOT EVENT__DISABLE_OPENSSL)
-    find_package(OpenSSL REQUIRED)
+    if (${CMAKE_VERSION} VERSION_GREATER "3.3")
+        find_package(OpenSSL REQUIRED COMPONENTS Crypto SSL)
+        list(APPEND LIB_APPS OpenSSL::SSL OpenSSL::Crypto)
+    else()
+        find_package(OpenSSL REQUIRED)
```

Comment:
Arguably better to do this based on behaviour - i.e. call find_package(OpenSSL ...), then if the modern-style targets exist use them otherwise use OPENSSL_LIBRARIES.